### ISSUE TITLE
Are most of the entries in the jacobian 0?

### ISSUE THREAD

**Issue by ConradMS** [dilevin/computer-graphics-kinematics#47]
Do most of the entires of the kinematics jacobian end up being zero? 

Because for any bone if we slightly increase the x or the z coordinate of its euler angle the tip of the bone and every other bone wont change position. Since both the x and z coords of the Euler angle of each bone correspond to a rotation around the x-axis (because xzx) and a rotation around the x-axis just twists the bone which does not move the tip of the bone, so it and every other bone doesn't end up moving?

so 2/3 of the entries of the Jacobian end up being 0?

Thanks.

**Comment by Zhecheng-Wang** [comment]
Yes, it should, and I think it might be more than 2/3. Jacobian entry $(i,j)$ basically describes how much a degree of freedom $i$ should change with respect to the change of another degree of freedom $j$. So if some degrees of freedom are not related (mathematically) in any way, their Jacobian entries will be $0$.
